### DISCUSSION TITLE
About MSNoise-tomo

### DISCUSSION THREAD

**Discussion by sujan1746** [ROBelgium/MSNoise#314] [Q&A] [answered]
Hi, actually, I have also same error with the below link.
I followed the process as your link until 'conda activate tomo' part. But when I try install using link related linux, but 'HTTP request sent, awaiting response... 410 Gone'. When I search the error, it has meaning of 'no more link exists'. Also I tried joining in the link but it seems very weird(any hint about msnoise-tomo) and don't think it is the right link.
So I think the link is expired. Could you please let me know what I have to do?

https://github.com/ROBelgium/MSNoise/issues/191#issue-530720505

**Comment by ThomasLecocq** [comment]
As explained elsewhere for the same errorw you have to install the package from source

**Reply by sujan1746** [reply]
About the source, do you mean the link and is it right that the link is related to JFrog(I think it is not)? When I install through the link, it says as I mentioned above, and when I click the link, it goes to JFrog site. Could you please check whether it is expired?

The link I used: http://msnoise.org/plugins/msnoise-tomo/msnoise_tomo-0.1b0-cp37-cp37m-linux_x86_64.whl
The link above link introduce: https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/
The figure I met:
![image](https://github.com/ROBelgium/MSNoise/assets/89066297/d9e3baf5-14a2-43a1-be4e-3b908924dede)

**Reply by ThomasLecocq** [reply, marked as answer]
No, those were pre built packages. You have to get the content of this GitHub repository and run python setup.py build & python setup.py install , or pip install .
